src/shell.zig

```diff
@@ -200,6 +222,70 @@ pub fn exec_stdout(shell: *Shell, comptime cmd: []const u8, cmd_args: anytype) !
     return res.stdout;
 }
 
+/// Spawns the process, piping its stdout and stderr.
+///
+/// The caller must `.kill()` and `.wait()` the child, to minimize the chance of process leak (
+/// sadly, child will still be leaked if the parent process is killed itself, because POSIX doesn't
+/// have nice APIs for structured concurrency).
+pub fn spawn(shell: Shell, comptime cmd: []const u8, cmd_args: anytype) !std.ChildProcess {
+    var argv = Argv.init(shell.gpa);
+    defer argv.deinit();
+
+    try expand_argv(&argv, cmd, cmd_args);
+
+    var child = std.ChildProcess.init(argv.items, shell.gpa);
+    child.stdin_behavior = .Ignore;
+    child.stdout_behavior = .Pipe;
+    child.stderr_behavior = .Pipe;
+
+    try child.spawn();
+
+    // XXX: child.argv is undefined when we return. This is fine though, as it is only used during
+    // `spawn`.
+    return child;
+}
+
+/// Runs the zig compiler.
+pub fn zig(shell: Shell, comptime cmd: []const u8, cmd_args: anytype) !void {
+    const zig_exe = try shell.project_root.realpathAlloc(
+        shell.gpa,
+        comptime "zig/zig" ++ builtin.target.exeFileExt(),
```

Comment:
> builtin.target.exeFileExt(),

TIL. Very nice.

Reply:
Does the path separator not need to be dependent on the OS? i.e. zig\zig.

Reply:
>TIL

Took me a while to hunt this one down! Obviously, there _should_ be some way to get the exe extention in the stdlib, but I had to grep my local copy of zig repo along and across to find it! `rg '".exe"'` did the trick

>Does the path separator not need to be dependent on the OS? i.e. zig\zig.

Windows APIs treat `/` and `\` the same. I _think_ `cmd.exe` might be confused by `\`, but I didn't check that .

Reply:
> Windows APIs treat / and \ the same. I think cmd.exe might be confused by \, but I didn't check that .

I was under the impression if you `exec` with `/` it will fail. Maybe I'm wrong. We'll see in CI :)

Reply:
Heh, turns out that Zig on windows breaks on `..`!

https://github.com/ziglang/zig/issues/16779

Reply:
>We'll see in CI :)

Seems ok!